### ISSUE TITLE
Feature request: Restart the Ruby process

### ISSUE THREAD

**Issue by suetanvil** [zenspider/enhanced-ruby-mode#195]
tl; dr, I would like `erm-reset` to be part of the public interface.  In addition, it might be nice to let me specify the Ruby command line to invoke as an optional argument.

The bigger picture is this: I use multiple versions and implementations of Ruby on various projects, often switching between them via `rbenv`. Emacs always uses the Ruby that's in the path when it starts up.  This is usually the system ruby which, once again, is the right thing **most** of the time.

However, I regularly find myself in the position of editing code with features that the Ruby belonging to `enh-ruby-mode` doesn't know about and that for one reason or another it's impractical (or at least high-friction) to change the default and restart everything. (Canonical example: I use endless function definitions; system ruby doesn't support those.)  The thing that works well **enough** for me is to set the appropriate Ruby in `rbenv`, then invoke `erm-reset`.  Of course, if I can't `mess` with `rbenv`, this can become more aggravating.

So it would be really helpful to me to have a command invoked with `M-x` that a) prompts for a path to Ruby, with `enh-ruby-program` as the default and then b) restarts the ripper process using the Ruby specified on the path.  This way, I can switch `enh-ruby-mode` to `/path/to/my/wierd/ruby` just for now and then switch it somewhere else later, all without losing my ludicrous number of open buffers and windows.

Anyway, thanks in advance for considering this.

**Comment by zenspider** [comment]
There's a [custom variable to specify which ruby](https://github.com/zenspider/enhanced-ruby-mode/blob/master/enh-ruby-mode.el#L194-L198) to run that you can tweak at any time (even temporarily) then you can call [`erm-reset`](https://github.com/zenspider/enhanced-ruby-mode/blob/master/enh-ruby-mode.el#L625). 

If you want to wrap that up in a function to help you choose which ruby and the like, you're welcome to do so, but I think that'd be rather specific to your situation.

**Comment by suetanvil** [comment]
What I mean by "public interface" is that the function be documented and follow the naming convention (`enh-ruby-*`) of those functions and variables that we expect end users to invoke.  Stuff that's semi-guaranteed not to change in breaking ways.  The name is important because a common pattern for discovery is to type `M-x` and the mode prefix and see what autocompletion brings up.

The reason I bring it up here instead of just hacking my own configuration is that this might be useful to other people so I thought I'd suggest it as an enhancement.

Actually, would you accept it as a pull request (assuming it's otherwise competently written, of course)?

It would:

1. Rename `erm-reset` to something like `enh-ruby-reset` with end-user-friendly documentation. 
2. Either extend it to prompt for a Ruby interpreter `enh-ruby-program` as the default (so the easiest use-case is to just hit enter) or add another function (e.g. `enh-ruby-reset-with-ruby`) that does the prompting and invokes `enh-ruby-reset`.

If so, I'd be willing to take a crack at it.
